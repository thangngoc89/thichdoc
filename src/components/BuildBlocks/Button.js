import React, { PropTypes as p } from "react";
import d from "../../design";
import cl from "classnames";

const Button = (
  {
    name,
    SVGIcon,
    block = false,
    size,
    bold = false,
    onButtonClick,
    className = ""
  }
) => {
  const aClassName = cl(
    "flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer",
    {
      [className]: true,
      "w-100": block,
      w4: !block,
      h3: size === "big",
      h2: size === "med",
      "b bw1": bold
    }
  );
  return (
    <a onClick={onButtonClick} className={aClassName}>
      <style jsx>
        {
          `
          a:hover > :global(svg), 
          a:focus > :global(svg) {
            fill: #fff;
          }
          `
        }
      </style>
      {SVGIcon && <SVGIcon className="w1" fill={d.colorPrimary} />}
      <span className="pl1">{name}</span>
    </a>
  );
};

Button.propsTypes = {
  action: p.string.isRequired,
  SVGIcon: p.object,
  size: p.string,
  block: p.bool,
  bold: p.bool,
  onButtonClick: p.func.isRequired
};

export default Button;
