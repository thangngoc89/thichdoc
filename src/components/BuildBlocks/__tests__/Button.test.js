import React from "react";
import Button from "../Button";
import renderer from "react-test-renderer";

const getTree = props => {
  const component = renderer.create(
    <Button name="Button" onButtonClick={jest.fn()} {...props} />
  );
  return component.toJSON();
};

test("Normal button", () => {
  expect(getTree()).toMatchSnapshot();
});
test("Bold button", () => {
  expect(getTree({ bold: true })).toMatchSnapshot();
});
test("Different size button", () => {
  expect(getTree({ size: "big" })).toMatchSnapshot();
});
test("Block button", () => {
  expect(getTree({ block: true })).toMatchSnapshot();
});
test("Custom class names button", () => {
  expect(getTree({ className: "foo" })).toMatchSnapshot();
});
test("All options button", () => {
  expect(
    getTree({ block: true, bold: true, size: "med", className: "foo" })
  ).toMatchSnapshot();
});

test("Trigger button onClick", () => {
  const spy = jest.fn();
  let tree = getTree({ onButtonClick: spy });
  expect(tree).toMatchSnapshot();
  tree.props.onClick();
  expect(spy.mock.calls.length).toBe(1);
});
