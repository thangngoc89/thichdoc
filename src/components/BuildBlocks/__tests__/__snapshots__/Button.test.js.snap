exports[`test All options button 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer foo w-100 h2 b bw1"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;

exports[`test Block button 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer  w-100"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;

exports[`test Bold button 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer  w4 b bw1"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;

exports[`test Custom class names button 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer foo w4"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;

exports[`test Different size button 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer  w4 h3"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;

exports[`test Normal button 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer  w4"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;

exports[`test Trigger button onClick 1`] = `
<a
  className="flex justify-center items-center pa2 ba b--red border-box f5 red hover-white hover-bg-red no-underline bg-animate pointer  w4"
  onClick={[Function]}>
  <style
    jsx={true}>
    
              a:hover > :global(svg), 
              a:focus > :global(svg) {
                fill: #fff;
              }
              
  </style>
  <span
    className="pl1">
    Button
  </span>
</a>
`;
